connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala

```diff
@@ -72,19 +79,19 @@ case class BigQueryWriteHelper(bigQuery: BigQuery,
       Runtime.getRuntime.addShutdownHook(createTemporaryPathDeleter)
 
       val format = options.intermediateFormat.dataSource
-      data.write.format(format).save(temporaryGcsPath.toString)
+      data.write.format(format).save(gcsPath.toString)
 
       loadDataToBigQuery
     } catch {
       case e: Exception => throw new RuntimeException("Failed to write to BigQuery", e)
     } finally {
-      cleanTempraryGcsPath
+      if (options.persistentGcsBucket.nonEmpty) cleanTempraryGcsPath
```

Comment:
Please restore the original line in order to keep a cleaner code. You can change the name to `cleanTempraryGcsPathIfNeeded` as it may not clean the path.

Reply:
Tks David.